### ISSUE TITLE
Make WebUI optional for leaner package dependencies

### ISSUE THREAD

**Issue by Lawouach** [locustio/locust#2318]
### Is your feature request related to a problem? Please describe.

Hi all,

Fantastic tool with clearly a lot of effort put into it :)

I am currently packaging it into a library at work and it works nicely. But I don't have a need for the WebUI.
The dependencies from locust web are therefore a bit dead weight if you don't mind me saying it that way.

- flask
- flask-basicauth
- flask-cors

Moreoever, at least flask-basicauth hasn't had a release for 8 years and might not respect the newer packaging requirements and doesn't create a wheel. 

### Describe the solution you'd like

Would it be possible to make the web interface optional so one could install locust without it at all?

### Describe alternatives you've considered

I thought of deleting them after the install but that would break as the web package is imported from various places in the code.

Cheers,

**Comment by cyberw** [comment]
Great idea! It could be done using extras https://peps.python.org/pep-0508/#extras

If someone were to make such a PR I’d definitely consider merging it.

**Comment by Lawouach** [comment]
Sweet. I'll see if I can find time to make one.

**Comment by cyberw** [comment]
Did you ever get around to looking at this?

**Comment by Lawouach** [comment]
Hello @cyberw 

I started some work but I couldn't find an easy way to isolate the web UI code. The `env` module imports the `WebUI` class for typing purpose and because that `env` module is quite central, it wasn't clear to me what to do there.

I'm not familiar enough with the codebase to make deep changes. I assume a blunt approach would be to something like `try: ... except ImportError:...` but it didn't feel elegant enough.

Unfortunately, I ran out of time to dig deeper. But I am still interested by that change so I may circle back to it someday.

Sorry for that :(
